File: src/main/java/com/itechart/demojavakotlin/controller/TicketController.java
package com.itechart.demojavakotlin.controller;

import com.itechart.demojavakotlin.entity.TicketEntity;
import com.itechart.demojavakotlin.service.TicketService;
import lombok.RequiredArgsConstructor;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.RestController;

import static org.springframework.http.HttpStatus.OK;

@RestController
@RequiredArgsConstructor
@RequestMapping("/tickets")
public class TicketController {

    private final TicketService ticketService;

    @PostMapping("/buy")
    public ResponseEntity<TicketEntity> buyTickets(@RequestParam final int quantity, @RequestParam final String movieName) {
        return new ResponseEntity<>(ticketService.buyTicket(quantity, movieName), OK);
    }
}

File: kotlinpath/demojavakotlin/src/main/kotlin/by/itechart/demojavakotlin/controller/TicketController.kt
package by.itechart.demojavakotlin.controller

import by.itechart.demojavakotlin.entity.TicketEntity
import by.itechart.demojavakotlin.model.TicketRequest
import by.itechart.demojavakotlin.service.TicketService
import org.springframework.http.HttpStatus
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping("/tickets")
class TicketController(private val ticketService: TicketService) {
    @PutMapping("/buy")
    fun buyTickets(@RequestBody quantity: Int): ResponseEntity<List<TicketEntity>> {
        return ResponseEntity<List<TicketEntity>>(ticketService.buyTicket(quantity), HttpStatus.OK)
    }

    @PostMapping
    fun addTicketsToMovie(ticketsRequest: TicketRequest): ResponseEntity<TicketEntity> {
        return ResponseEntity<TicketEntity>(ticketService.addTicketsToMovie(ticketsRequest), HttpStatus.OK)
    }

    @DeleteMapping("/{movieName}")
    fun removeTicketsByMovieName(@PathVariable movieName: String): ResponseEntity<Void> {
        ticketService.removeTicketsByMovieName(movieName)
        return ResponseEntity(HttpStatus.NO_CONTENT)
    }

}